### DISCUSSION TITLE
How t make recursive models.

### DISCUSSION THREAD

**Discussion by aliscie** [fastapi/fastapi#6293] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class File(Base):
    __tablename__ = "files"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String)
    content = Column(String)
    time_created = Column(DateTime(timezone=True), server_default=func.now())


    parent = relationship("File", back_populates="children")
    children = relationship("FileChild", back_populates="parent")


my schema
```graphql
extend type Mutation {
    file(name: String!, content: String, parent_id:Int): String!
}

type File {
    name: String
    children: [File]
}

extend type Query {
    files : [File]
}


```
```


### Description

Hello, I want to do recursive model like this, but is this the right way t do it? also, I am using ariadne graphql, so i should make specific schema.graphql not sure how to do that yet.


### Operating System

Linux, Windows, macOS

### Operating System Details

_No response_

### FastAPI Version

any

### Python Version

any

### Additional Context

_No response_

**Comment by JarroVGIT** [comment, marked as answer]
I am not sure how this is related to FastAPI. Maybe this is better suited for Stackoverflow?

**Comment by imacat** [comment]
I suppose this is a question to SQLAlchemy, not FastAPI.
https://www.sqlalchemy.org/support.html

**Comment by aliscie** [comment]
x
